plugins/chat/assets/javascripts/discourse/initializers/chat-setup.js

```diff
@@ -30,6 +32,13 @@ export default {
     }
 
     withPluginApi("0.12.1", (api) => {
+      api.onPageChange((path) => {
+        const route = this.router.recognize(path);
+        if (route.name.startsWith("chat.")) {
+          this.chatHistory.visit(route);
+        }
+      });
```

Comment:
I will investigate if I can't put this at directly in the router, but this might be complicated with the drawer